File: app/Http/Controllers/SpecialistPatternsJava/Creational/AbstractFactory/AbstractFactoryJava/src/roofs/WoodRoof.java
package roofs;

import static java.lang.System.out;

// Класс WoodRoof представляет крышу с деревянной отделкой.
public class WoodRoof implements Roof {

    // Метод cover() выполняет покрытие крыши деревянной отделкой.
    @Override
    public Roof cover() {
        out.println("Покрытие крыши деревянной отделкой");
        return this;
    }

    // Метод waterProtect() обеспечивает защиту крыши от воды.
    @Override
    public void waterProtect() {
        out.println("Защита крыши от воды");
    }
}

File: app/Http/Controllers/SpecialistPatterns/AbstractFactory/Roofs/WoodRoof.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\AbstractFactory\Roofs;

// Класс WoodRoof представляет крышу с деревянной отделкой.
use Illuminate\Support\Facades\Log;

class WoodRoof implements Roof
{

    // Метод cover() выполняет покрытие крыши деревянной отделкой.
    public function cover(): Roof
    {
        Log::channel('sergey')->info("Покрытие крыши деревянной отделкой");
        return $this;
    }

    // Метод waterProtect() обеспечивает защиту крыши от воды.
    public function waterProtect(): void
    {
        Log::channel('sergey')->info("Защита крыши от воды");
    }
}
